apps/calendar/test/marionette/day_view_test.js

```diff
@@ -327,6 +327,38 @@ marionette('day view', function() {
     });
   });
 
+  test('double tap all day + toggle all day', function() {
+    day.waitForHourScrollEnd();
+
+    day.actions
+      .doubleTap(day.activeAllDays[0], 150, 30)
+      .perform();
+
+    var event = app.editEvent;
+    event.waitForDisplay();
+    assert.ok(event.allDay, 'is all day event');
+    var oldStart = event.startTime;
+    var oldEnd = event.endTime;
+    event.allDay = false;
+    event.title = 'Foo';
+    client.waitFor(function() {
+      return event.startTime !== oldStart && event.endTime !== oldEnd;
+    });
+    var now = new Date();
+    var start = new Date(now.getTime());
+    start.setHours(now.getHours() + 1, 0, 0, 0);
+    var end = new Date(start.getTime());
+    end.setHours(start.getHours() + 1, 0, 0, 0);
+    var isToday = start.toISOString().slice(0, 10) === event.startDate;
+    var expectedStart = isToday ? pad(start.getHours()) + ':00:00' : '08:00:00';
+    var expectedEnd = isToday ? pad(end.getHours()) + ':00:00' : '09:00:00';
+    assert.equal(event.startTime, expectedStart, 'startTime');
+    assert.equal(event.endTime, expectedEnd, 'endTime');
+
+    event.save();
+    day.waitForDisplay();
```

Comment:
we already have other tests that checks if saved events are displayed on day/week/month/read_event views, so this test is enough to check if the bug was fixed (adding more steps just increases the chance of potential intermittent failures...) - this test is already too complex because of the time handling..